_sections/91-encrypted-forms.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Encryption 
+---
+Encrypted forms provide a mechanism to keep your data private even when using http: for communications (e.g., when you do not have an SSL certificate or https: is not available).
+
+### Form Definition
+
+Here is an excerpt used in an encrypted form:
+
+{% highlight xml %}
+<submission method="form-data-post"
+                action="https://my-opendatakit.appspot.com/submission"
+                base64RsaPublicKey="MIIBIjANB...JCwIDAQAB" />
+{% endhighlight %}
+
+The form encryption is governed by the settings on the [Settings Worksheet](https://opendatakit.org/help/form-design/xlsform/#settings_ws).Encrypted forms must specify a *submission_url* and a *public_key* on this worksheet.If both are specified, XLSForm will generate a encrypted-form definition.Encrypted form definitions must include a **unique OpenRosa instanceID** and have an explicit `<submission/>` element.Within this tag, the method attribute should always be `form-data-post`. The action attribute should be the url to which the submission should be posted; this is the server url. Finally, what identifies the form as an encryted form is the presence of a *base64RsaPublicKey* attribute. This should be the base64 encoding of the RSA public key that client uses to encrypt the symmetric encryption key it creates to encrypt a finalized instance of this form (a different symmetric encryption key is created for every finalized form).
```

Comment:
We keep XLSForm references out of this spec since XLSForm is just one way of generating XForms, and to avoid confusion. So only describe `<submission>`, `method`, `action`, `base64RsaPublicKey` here.

Reply:
**unique OpenRosa instanceID* is required for all forms, so probably not needed here.

Reply:
Yes. Additionally, I would move the narrative before the example. I would go with something like:

Encrypted form definitions must have an explicit `<submission />` element with the following attributes:
- `method`: always set to `form-data-post`
- `action`: the url of the server to send this submission to
- `base64RsaPublicKey`: base64 encoding of an RSA public key. The corresponding private key will be needed to decrypt submissions. 

The client generates a different symmetric encryption key for each finalized form and uses it to encrypt the submission and all media files. The `base64RsaPublicKey` is used to encrypt the symmetric key with the RSA/NONE/OAEPWithSHA256AndMGF1Padding algorithm. 

===

I'm not sure about the last paragraph as it's starting to get into non-XForms details. Perhaps it should live in the API docs which would also describe the manifest.

@Martijn, do you think `method` should be part of this spec? Is it likely to have any other valid values? W3C XForms uses it for HTTP method as described in https://www.ibm.com/support/knowledgecenter/SSS28S_8.2.0/XFDL_Specification/xfdl_r_xforms_submission.html But I don't think any tool in the ODK ecosystem supports those.

Reply:
I think `method` should be in there (but it should actually have value `"post"` in hindsight). Your question leads me to suspect that ODK Collect is not actually reading/using this attribute. In that case, we could change it to `"post"` perhaps....

The last paragraph (about the encryption algorithm) is good to include here, I think.